### ISSUE TITLE
[Blazor]Error MSB4018: The "ResolveBlazorRuntimeDependencies" task failed unexpectedly.

### ISSUE THREAD

**Issue by ZhangGaoxing** [dotnet/aspnetcore#21046]
### Describe the bug
I referenced my own .net standard 2.1 nuget package in the Blazor web assembly project. But it works well in the Blazor server side project.

```
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018: The "ResolveBlazorRuntimeDependencies" task failed unexpectedly.
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018: System.InvalidOperationException: Multiple assemblies found with the same assembly name 'BouncyCastle.Crypto':
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018: BouncyCastle.Crypto
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018: C:\Users\zhang\.nuget\packages\portable.bouncycastle\1.8.5\lib\netstandard2.0\BouncyCastle.Crypto.dll
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018:    在 Microsoft.AspNetCore.Components.WebAssembly.Build.ResolveBlazorRuntimeDependencies.<ResolveRuntimeDependenciesCore>g__CreateAssemblyLookup|17_1(IEnumerable`1 assemblyPaths)
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018:    在 Microsoft.AspNetCore.Components.WebAssembly.Build.ResolveBlazorRuntimeDependencies.ResolveRuntimeDependenciesCore(String entryPoint, IEnumerable`1 applicationDependencies, IEnumerable`1 monoBclAssemblies)
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018:    在 Microsoft.AspNetCore.Components.WebAssembly.Build.ResolveBlazorRuntimeDependencies.Execute()
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018:    在 Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(324,5): error MSB4018:    在 Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

### Further technical details
```
PS C:\Users\zhang> dotnet --info
.NET Core SDK（反映任何 global.json）:
 Version:   3.1.201
 Commit:    b1768b4ae7

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.201\

Host (useful for support):
  Version: 3.1.3
  Commit:  4a9f85e9f8

.NET Core SDKs installed:
  3.1.100 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by ZhangGaoxing** [comment]
I tried this https://github.com/dotnet/aspnetcore/issues/17644#issuecomment-563355603 but it didn't work

**Comment by ZhangGaoxing** [comment]
If I add `<BlazorWebAssemblyEnableLinking>true</BlazorWebAssemblyEnableLinking>` to csproj file, the error is as follows
```
3>Fatal error in Mono IL Linker
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(274,5): error : Unhandled exception. Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'System.Diagnostics.PerformanceCounter, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
3> ---> Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'System.Diagnostics.PerformanceCounter, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
3>   at Mono.Cecil.BaseAssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
3>   at Mono.Linker.AssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
3>   at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
3>   at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
3>   at Mono.Linker.LinkContext.ResolveReferences(AssemblyDefinition assembly)
3>   at Mono.Linker.Steps.LoadReferencesStep.ProcessReferences(AssemblyDefinition assembly)
3>   at Mono.Linker.Steps.LoadReferencesStep.ProcessAssembly(AssemblyDefinition assembly)
3>   at Mono.Linker.Steps.BaseStep.Process(LinkContext context)
3>   at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
3>   at Mono.Linker.Pipeline.Process(LinkContext context)
3>   at Mono.Linker.Driver.Run(ILogger customLogger)
3>   at Mono.Linker.Driver.Execute(String[] args, ILogger customLogger)
3>   at Mono.Linker.Driver.Main(String[] args)
3>C:\Users\zhang\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\3.2.0-preview4.20210.8\targets\Blazor.MonoRuntime.targets(274,5): error : ILLink failed with exit code -532462766.
```

**Comment by javiercn** [comment]
@ZhangGaoxing thanks for contacting us.

Can you provide a minimal repro project that demonstrates the issue so that we can look at what is going on?

**Comment by ZhangGaoxing** [comment]
I analyzed my NuGet package references and found that there were two `bouncycastle.crypto.dll`. But interestingly, no matter in Core MVC or Blazor Server Side, there is no error. Now I downgraded MySQL package to solve this problem.

![1](https://user-images.githubusercontent.com/15957347/79839982-dad07500-83e7-11ea-8663-084c75232046.png)
![2](https://user-images.githubusercontent.com/15957347/79839986-dc9a3880-83e7-11ea-8f84-450c4607fdaf.png)
